api/storage_service.cc

```diff
@@ -488,9 +488,7 @@ void set_storage_service(http_context& ctx, routes& r, sharded<service::storage_
         return scylla_version();
     });
 
-    ctx.db.local().get_builtin_routine_registry().register_routine("scylla_version",
-            db::make_single_value_routine<sstring>("scylla_version", [](db::builtin_routine::parameter_map&& par) {
-        return make_ready_future<sstring>(scylla_version());}));
+    db::add_system_function("scylla_version", [] { return make_ready_future<sstring>(scylla_version()); });
```

Comment:
Actually, this one doesn't `require_thread` and indeed `is_pure`.
Also, I don't understand why this particular function is not put into `builtin_routines`?

Reply:
`add_system_function` registers a function as a CQL one, so it can be used inside statements without the need for long CALL or SELECT syntax, i.e:

```CQL
cqlsh> INSERT INTO ks.versions (p, k, v) VALUES (1, 1, scylla_version());
cqlsh> SELECT * FROM ks.versions ;

 p | k | v
---+---+------------------------------
 1 | 1 | 4.7.dev-0.20211215.d0cdfb30d

(1 rows)

```

Reply:
Ah, now I see.
So, more or less, CALL syntax is for maintenance operations; DUAL syntax is for "asking questions to the DB". Great, if that's the industry practice then OK.